### ISSUE TITLE
Better error message for permission denied errors in dials.import

### ISSUE THREAD

**Issue by rjgildea** [dials/dials#495]
Please could dials.import check for permissions problems on input files, and return a better error in such cases:
```
$ dials.image_viewer datablock.json 
Sorry: Unable to handle the following arguments:
  datablock.json failed during DataBlock processing:
    Traceback (most recent call last):
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/dials/util/options.py", line 310, in try_read_datablocks
        self.datablocks.append(converter.from_string(argument))
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/dials/util/phil.py", line 47, in from_string
        check_format=self._check_format))
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/datablock.py", line 991, in from_json_file
        return DataBlockFactory.from_json(infile.read(), check_format)
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/datablock.py", line 981, in from_json
        check_format)
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/datablock.py", line 969, in from_dict
        importer = DataBlockDictImporter(obj, check_format)
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/datablock.py", line 673, in __init__
        self.datablocks = self._load_datablocks(obj, check_format)
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/datablock.py", line 688, in _load_datablocks
        return [self._load_datablocks(dd, check_format) for dd in obj]
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/datablock.py", line 738, in _load_datablocks
        format_kwargs=format_kwargs)
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/imageset.py", line 525, in make_sweep
        format_class = Registry.find(filenames[0])
      File "/dls/mx-scratch/rjgildea/software/cctbx/modules/cctbx_project/dxtbx/format/Registry.py", line 83, in find
        raise IOError('no format support found for %s' % image_file)
    IOError: no format support found for /dls/science/users/xyz/multiaxis/thau_1/thau_1_10_0001.cbf

$ ls -l /dls/science/users/xyz/thau_1/thau_1_10_0001.cbf
ls: cannot access /dls/science/users/xyz/thau_1/thau_1_10_0001.cbf: Permission denied
```

**Comment by Anthchirp** [comment]
Root cause for this is probably the current logic of "exception means wrong class", ie. #335

**Comment by ndevenish** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Related: #495 Better error message for permission denied errors in dials.import](https://trello.com/c/kamWgPeY/24-related-495-better-error-message-for-permission-denied-errors-in-dialsimport)

**Comment by ndevenish** [comment]
Fixed with cctbx/cctbx_project@0970a2d3809567dac4e42c5fea68ffd42ae96a40, although the underlying issues in #335 remain. Now looks like:
```
Sorry: Unable to handle the following arguments:
  "datablock.json" failed during DataBlock processing:
    Permission denied: /path/to/data/transthyretin_h5/200Hz_0.1dpf_1_master.h5
```
